### ISSUE TITLE
Add function player(playerName: string)

### ISSUE THREAD

**Issue by ChrisTechke** [SkriptLang/Skript#341]
Server Version: CraftBukkit version git-Spigot-7d78b81-775f1e1 (MC: 1.11.2) (Implementing API version 1.11.2-R0.1-SNAPSHOT)

Skript Version: dev23 (https://github.com/bensku/Skript/releases/tag/dev23)

The Skript below works fine

```
on join:
	if player is in world "world":
		message "Test" to player
```

However, if you change the world name to anything but a default Minecraft world name it will not work.

Skript I am testing:

```
on join:
	if player is in world "TestWorld":
		message "Test" to player
```

Console output when reloading:

Can't compare a player with 'in world "TestWorld"' (Test.sk, line 2: if player is in world "TestWorld":')

**Comment by JeepSmash** [comment]
I have no errors loading this code if the named world is an actual world on the server.  I get 
`Can't compare a player with 'in world "TestWorld"' (Test.sk, line 2: if player is in world "TestWorld":')`
if i execute the code exactly as you have it but if i change "TestWorld" to an actual world on my server "minigames" then the code loads and executes without error.  Maybe your spelling/capitalization of your world is not correct.

**Comment by ChrisTechke** [comment]
I have the world loaded on my server with the other worlds and my spelling and punctuation are perfect with them both, but I still receive an error.

**Comment by Pyon-Pyon** [comment]
Someone had this issue too, this should be working:
```
player is in world("world")
```
but surely it's a bug

**Comment by TheBentoBox** [comment]
I actually just tested this on dev23 with the latest Spigot and am not having this issued. Used the following code:

```
on right click:
	if player is in world "test":
		broadcast "Clicked in %world%"
	else if player is in world "world":
		broadcast "Clicked in %world%"
on join:
	if player is in world "test":
		broadcast "Joined in %world%"
	else if player is in world "world":
		broadcast "Joined in %world%"
```

Had worlds "world" and "test" up on my server, both events fired properly in both worlds, e.g. [this](https://gyazo.com/99adfefa22cd7f940475211aa49306ad). Are you absolutely sure you were in the right world?

Could anyone else (@Pyon-Pyon?) test this to verify for me if it's also an issue for them?

**Comment by ChrisTechke** [comment]
@TheBentoBox 

Yeah, I was in the right world. I had the TestWorld on my server too.

Perhaps it's a bug for Java 7 users? I'm not sure.

@Pyon-Pyon Thanks. It works if I use that.

**Comment by Pyon-Pyon** [comment]
@TheBentoBox the condition is working fine for me in the latest Spigot and with the latest release too, though, more than one person asked me due to this problem and what I said above fixed that, weird...

**Comment by TheBentoBox** [comment]
We'll see what bensku wants to do about this considering it seems to work for a lot of people and not for others. Not sure if it's really a bug or an issue on the user's end.

**Comment by bensku** [comment]
TODO: add function player(playerName: string)

It should help for people who have worlds that share names with some players.

**Comment by TheBentoBox** [comment]
Updating this old issue's title to better reflect what Bensku wanted done here.

**Comment by Syst3ms** [comment]
I don't understand how this function would help, would anyone mind explaining ?

**Comment by TheBentoBox** [comment]
I'll be honest, I wasn't sure how it related to the issue at hand either, but I think it'd make sense to add regardless. Functions look prettier than `parsed as player` (IMO, at least).

**Comment by TheLimeGlass** [comment]
This isn't a player issue. The reason this condition isn't working is because the world wasn't loaded. Any world needs to be loaded and able to be found under Bukkit's `getWorld(string)` or `getWorlds()` when the server starts.

This has been a common issues for years. The main reason why I made https://github.com/TheLimeGlass/skript-world so that worlds can be handled after the server has already been started.

Basically in conclusion if you're still confused. Skript keeps in mind all the worlds that have been loaded on start. When a new world is created/loaded as the server is running, Skript doesn't track that world until the server has been restarted again. Any conditions/effects/expressions that call on a world that isn't in the cache will error.

**Comment by Wealthyturtle** [comment]
Closing this issue as this is a world loading problem, in favor of #1144
